src/TSHelper.ts

```diff
@@ -316,4 +316,18 @@ export class TSHelper {
         return defaultArrayPropertyNames.has(methodName);
     }
 
+    public static isNonFalsible(type: ts.Type): boolean {
+        const falsibleFlags = ts.TypeFlags.Boolean
+            | ts.TypeFlags.BooleanLiteral
+            | ts.TypeFlags.Undefined
+            | ts.TypeFlags.Any;
```

Comment:
Will this catch null-able types?
```ts
declare function foo(): string | null;
const x = condition ? foo() : "bar";
```

Reply:
Update: now the above only applies when strictNullChecks are not enabled. Otherwise I recursively iterate through all union types and check if they are all non-falsible.

Reply:
Re-reading this I realize why I got confused. Will a null literal get caught by those flags?

Reply:
Good point, I wasn't aware that there is a difference between `null` and `undefined`. I added an additional check for it.